client/src/components/ListGrid/ListGrid.js

```diff
@@ -0,0 +1,258 @@
+import React, { useState } from 'react';
+import './ListGrid.scss';
+import { Edit, Verified, GppBad } from '@mui/icons-material';
+
+function sortList(data, sortColumn, sortOrderAsc = true) {
+  return [...data].sort((a, b) => {
+    const aa = a[sortColumn];
+    const bb = b[sortColumn];
+
+    if (aa === bb) return 0;
+    if (aa === null || aa === undefined) return sortOrderAsc ? 1 : -1;
+    if (bb === null || bb === undefined) return sortOrderAsc ? -1 : 1;
+
+    // handle boolean value sorting for the "broken" column in the Source page
+    if (typeof aa === 'boolean' && typeof bb === 'boolean') {
+      return sortOrderAsc ? aa - bb : bb - aa;
+    }
+
+    const aStr = String(aa).toLowerCase();
+    const bStr = String(bb).toLowerCase();
+
+    if (sortOrderAsc) {
+      // If ascending, compare the strings and return either -1 or 1 based on the result.
+      return aStr < bStr ? -1 : 1;
```

Comment:
Not sure if this is a concern, but it might better to use `localeCompare` here. We don't care about case sensitivity or other languages, but there could be some edge cases where accents, punctuation, or numbers cause it to sort in unexpected ways.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare